src/libsync/syncengine.cpp

```diff
@@ -329,6 +329,51 @@ void SyncEngine::deleteStaleErrorBlacklistEntries(const SyncFileItemVector &sync
     _journal->deleteStaleErrorBlacklistEntries(blacklist_file_paths);
 }
 
+void SyncEngine::conflictRecordMaintenance()
+{
+    QByteArrayList conflictRecordPaths;
+
+    // Remove stale conflict entries from the database
+    if (account()->capabilities().uploadConflictFiles()) {
+        // If conflicts are uploaded, we just wipe all entries that no
+        // longer have a corresponding metadata entry.
+        _journal->deleteConflictRecordsNotInMetadata();
+        conflictRecordPaths = _journal->conflictRecordPaths();
+    } else {
+        // Otherwise check which conflict files are still existant
+        // and delete stale records.
+        conflictRecordPaths = _journal->conflictRecordPaths();
+        for (const auto &path : conflictRecordPaths) {
+            auto fsPath = _propagator->getFilePath(QString::fromUtf8(path));
+            if (!QFileInfo(fsPath).exists()) {
+                _journal->deleteConflictRecord(path);
+            }
+        }
+    }
+
+    // Did the sync see any conflict files that don't yet have records?
+    // If so, add them now.
+    //
+    // This happens when the conflicts table is new or when conflict files
+    // are downlaoded but the server doesn't send conflict headers.
+    for (const auto &path : _seenFiles) {
+        auto bapath = path.toUtf8();
```

Comment:
(Would be nice: have a Utility::isConflictFile that works with QString and avoid this allocation, _seenFiles can be big)